src/node-lib/curriculum-api-2023/queries/subjectListing/processLesson.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  subjects,
+  subjectSlugs,
+  SyntheticUnitvariantLessons,
+} from "@oaknational/oak-curriculum-schema";
+
+interface UnprocessedSubject {
+  [key: string]: {
+    subjectTitle: typeof subjects | string;
+    subjectSlug: typeof subjectSlugs | string;
+    programmeSlug: string;
+    unitIds: Set<number>;
+    lessonSlugs: Set<string>;
+    programmeSlugs: Set<string>;
+  };
+}
+
+interface ProcessedSubject {
+  subjectTitle: typeof subjects | string;
+  subjectSlug: typeof subjectSlugs | string;
+  programmeSlug: string;
+  unitCount: number;
+  lessonCount: number;
+  programmeCount: number;
+}
+
+export const processLessons = (
+  lessons: SyntheticUnitvariantLessons[],
+): ProcessedSubject[] => {
+  const subjects = {} as UnprocessedSubject;
+
+  for (const lesson of lessons) {
```

Comment:
this looks like it should be a reduce `lessons.reduce(()=>  ..., {} as Record<string, Subject>)`